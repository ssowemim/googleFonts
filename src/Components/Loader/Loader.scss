.Loader {
  position: fixed;
  left: 0;
  top: 0;
  height: 100%;
  width: 100%;

  .Loader__Overlay {
    width: 100%;
    height: 100%;
    background: rgba(0, 0, 0, .4);
    position: fixed;
    top: 0;
    bottom: 0;
    left: 0;
    right: 0;
  }

  .Loader__Wrapper {
    width: 50px;
    height: 50px;
    display: flex;
    flex-wrap: wrap;
    position: absolute;
    top: 50%;
    left: 50%;
    transform: translateX(-50%) translateY(-50%)
  }

  .Loader__section {
    width: 25px;
    height: 25px;
    animation: loading 1s linear infinite;
  }

  .Loader__section:nth-child(1) {
    background: #FFDE00;
    animation-delay: 0;
  }

  .Loader__section:nth-child(2) {
    background: #000;
    animation-delay: 0.25s;
  }

  .Loader__section:nth-child(3) {
    background: #000;
    animation-delay: .75s;
  }

  .Loader__section:nth-child(4) {
    background: #FFDE00;
    animation-delay: .50s;
  }
}

@keyframes loading {
  0% {
    opacity: 0.4;
  }
  100% {
    opacity: 1;
  }
}
 