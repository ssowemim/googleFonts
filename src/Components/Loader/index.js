import React, { Component } from 'react';
import PropTypes from 'prop-types';
import './Loader.css';

class Loader extends Component {
    render() {
        return (
            <div className="Loader">
                <div className="Loader__Overlay"></div>
                <div className="Loader__Wrapper">
                    <div className="Loader__section"></div>
                    <div className="Loader__section"></div>
                    <div className="Loader__section"></div>
                    <div className="Loader__section"></div>
                </div>
            </div>
        );
    }
}

Loader.propTypes = {

};

export default Loader;