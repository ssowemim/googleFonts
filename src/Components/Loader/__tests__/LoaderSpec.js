import React from 'react';
import renderer from 'react-test-renderer';
// import { shallow } from 'enzyme';
import Loader from './../index';

describe('ColorPicker', () => {

    it('renders correctly', () => {
        let tree = renderer.create(
            <Loader />
        );
        expect(tree.toJSON()).toMatchSnapshot();
    });

});
