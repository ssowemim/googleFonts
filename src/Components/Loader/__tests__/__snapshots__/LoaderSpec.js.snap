// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorPicker renders correctly 1`] = `
<div
  className="Loader"
>
  <div
    className="Loader__Overlay"
  />
  <div
    className="Loader__Wrapper"
  >
    <div
      className="Loader__section"
    />
    <div
      className="Loader__section"
    />
    <div
      className="Loader__section"
    />
    <div
      className="Loader__section"
    />
  </div>
</div>
`;
